src/stub/renderer/language.rs

```diff
@@ -0,0 +1,159 @@
+            (true, Some(false)) => variable_name.to_lowercase(),
+            (true, _) => variable_name.to_string(),
+            (false, _) => self.variable_format.convert(variable_name),
+        };
+
+        self.escape_keywords(converted_variable_name)
+    }
+
+    pub fn escape_keywords(&self, variable_name: String) -> String {
+        if self.keywords.contains(&variable_name) {
+            format!("_{variable_name}")
+        } else {
+            variable_name
+        }
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(rename_all = "PascalCase")]
+pub struct TypeTokens {
+    pub int: Option<String>,
+    pub float: Option<String>,
+    pub long: Option<String>,
+    pub bool: Option<String>,
+    pub word: Option<String>,
+    pub string: Option<String>,
+}
+
+impl TryFrom<&str> for Language {
+    type Error = anyhow::Error;
+
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        let lang_folders: Vec<String> = std::fs::read_dir("config/stub_templates/")
+            .context("Should have stub_templates directiory")?
+            .map(|read_dir| read_dir.unwrap().file_name().into_string().expect("Template path must be UTF"))
+            .collect();
+
+        Self::find_lang_by_name(&value.to_lowercase(), &lang_folders)?
+            .or(Self::find_lang_by_alias(&value.to_lowercase(), &lang_folders))
+            .ok_or(anyhow!("Unsupported language: {}", value))
+    }
+}
+
+impl Language {
+    pub fn template_glob(&self) -> String {
+        format!("config/stub_templates/{}/*.{}.jinja", self.name, self.source_file_ext)
+    }
+
+    fn find_lang_by_name<'a>(name: &'a str, lang_folders: &'a [String]) -> Result<Option<Language>> {
+        if lang_folders.iter().any(|l| l == name) {
+            let language_config_filepath = format!("config/stub_templates/{}/stub_config.toml", name);
+            let config_file_content = fs::read_to_string(language_config_filepath)
+                .context(format!("No stub configuration exists for {}", name))?;
+
+            Ok(toml::from_str(&config_file_content)
+                .context("There was an error loading the stub configuration")?)
+        } else {
+            Ok(None)
+        }
+    }
+
+    fn find_lang_by_alias<'a>(name: &'a str, lang_folders: &'a [String]) -> Option<Language> {
+        lang_folders
+            .iter()
+            .filter_map(|folder| {
+                let language_config_filepath = format!("config/stub_templates/{}/stub_config.toml", folder);
+                match fs::read_to_string(language_config_filepath) {
+                    Ok(config_file_content) => toml::from_str::<Language>(&config_file_content).ok(),
+                    _ => None,
+                }
+            })
+            .find(|l| match &l.aliases {
+                Some(aliases) => aliases.contains(&name.to_string()),
+                None => false,
+            })
+    }
```

Comment:
Here the choice was that invalid toml would not be fatal, the app simply continues looking for a match for the alias, if they all fail then we return an error message.

If you are making a new language toml, you should first try gen with the language name, not with an alias, in which case you would run into:
```rust
            Ok(toml::from_str(&config_file_content)
                .context("There was an error loading the stub configuration")?)
```

It didn't make sense to me to raise an error when any language toml was invalid, since the command doesn't expressly state that it verifies that all configuration is valid.

If you disagree, I can change it.